PROJECTS/SDSS/PS1QLS.py

```diff
@@ -121,7 +121,44 @@ def make_roc(self,truth):
         plt.plot(fpr,tpr,'b--')
         plt.xlabel('FPR')
         plt.ylabel('TPR')
-        return fpr,tpr
+        return fpr, tpr
+    
+    def make_pc_plot(self, truth, abundance=1e-3):
+        """
+        Plots purity against completeness, assuming a natural abundance of lenses.
+        
+        Parameters
+        ----------
+        truth: ndarray
+            True labels for the test dataset objects.
+        abundance: float
+            Fraction of all objects in the expected parent dataset that are actually lenses.
+        // code starts here
+        
+        precision, recall, _ = sklearn.metrics.precision_recall_curve(truth, self.lQSO_probs, pos_label=1)
+        fpr, tpr, _ = metrics.roc_curve(truth,self.lQSO_probs,pos_label=1)
```

Comment:
Thanks! In fact, because we imported metrics from sklearn, the first lines should be `metrics.precision_recall_curve.` The other problem is, as you pointed out, those functions return arrays that have different elements. I will try to fix it by finding a way to give same threshold points to plot.